packages/plugins/typed-variable-modal/src/Modal.js

```diff
@@ -0,0 +1,378 @@
+  }
+
+  /**
+   * Create all the dom elements for the modal.
+   */
+  render() {
+    /*
+     * Creates the Modal. The generated modal looks like:
+     * <div class="blockly-modal-container" role="dialog">
+     *   <header class="blockly-moddal-header">
+     *     <h2 class="blockly-modal-header-title">Modal Name</h2>
+     *     <button class="blockly-modal-btn">X</button>
+     *   </header>
+     *   <div class="blockly-modal-content">
+     *   </div>
+     *   <div class="blockly-modal-footer">
+     *     <button class="blockly-modal-btn blockly-modal-btn-primary">OK</button>
+     *     <button class="blockly-modal-btn">Cancel</button>
+     *   </div>
+     * </div>
+     */
+
+    // Create Container
+    this.htmlDiv_ = document.createElement('div');
+    this.htmlDiv_.className = 'blockly-modal-container';
+    this.htmlDiv_.setAttribute('role', 'dialog');
+    // End creating the container
+
+    // Create the header
+    const modalHeader = document.createElement('header');
+    modalHeader.className = 'blockly-modal-header';
+
+    this.renderHeader_(modalHeader);
+
+    const exitButton = document.createElement('button');
+    exitButton.className = 'blockly-modal-btn blockly-modal-btn-close';
+    this.addEvent_(exitButton, 'click', this, this.onCancel_);
+    modalHeader.appendChild(exitButton);
+    // End create header
+
+    // Create content
+    const modalContent = document.createElement('div');
+    modalContent.className = 'blockly-modal-content';
+    this.renderContent_(modalContent);
+    // End creating content
+
+    // Create Footer
+    const modalFooter = document.createElement('footer');
+    modalFooter.className = 'blockly-modal-footer';
+
+    this.renderFooter_(modalFooter);
+    // End creating footer
+
+    this.htmlDiv_.appendChild(modalHeader);
+    this.htmlDiv_.appendChild(modalContent);
+    this.htmlDiv_.appendChild(modalFooter);
+  }
+
+  /**
+   * Render content for the modal header.
+   * @param {HTMLElement} headerContainer The modal's header div.
+   * @protected
+   */
+  renderHeader_(headerContainer) {
+    const modalTitle = document.createElement('h2');
+    modalTitle.className = 'blockly-modal-header-title';
+    modalTitle.appendChild(document.createTextNode(this.title_));
+    this.htmlDiv_.setAttribute('aria-labelledby', this.title_);
+    headerContainer.appendChild(modalTitle);
+  }
+
+  /**
+   * Render content for the modal content div.
+   * @param {HTMLDivElement} _contentContainer The modal's content div.
+   * @protected
+   */
+  renderContent_(_contentContainer) {
+    // No-op on the base class.
+  }
+
+  /**
+   * Render content for the modal footer.
+   * @param {HTMLElement} _footerContainer The modal's footer div.
+   * @protected
+   */
+  renderFooter_(_footerContainer) {
+    // No-op on the base class.
+  }
+}
+Blockly.Css.register([`
+    .blockly-modal-container {
+      background-color: white;
+      border: 1px solid gray;
+      max-width: 50%;
```

Comment:
why set max-width to 50%, and then width to 400px. That doesn't make sense to me?

Reply:
whoops, left over from a previous version. Removed.